src/main/java/com/thealgorithms/sorts/RadixSort.java

```diff
@@ -2,61 +2,58 @@
 
 import java.util.Arrays;
 
-final class RadixSort {
+/**
+ * This class provides an implementation of the radix sort algorithm.
+ * It sorts an array of integers in increasing order.
+ */
+public final class RadixSort {
     private RadixSort() {
     }
 
-    private static int getMax(int[] arr, int n) {
-        int mx = arr[0];
-        for (int i = 1; i < n; i++) {
-            if (arr[i] > mx) {
-                mx = arr[i];
-            }
+    /**
+     * Sorts an array of integers using the radix sort algorithm.
+     *
+     * @param array the array to be sorted
+     * @return the sorted array
+     */
+    public static int[] sort(int[] array) {
+        if (array.length == 0) {
+            return array;
         }
-        return mx;
+
+        final int max = Arrays.stream(array).max().getAsInt();
+        for (int exp = 1; max / exp > 0; exp *= 10) {
```

Comment:
Please consider using [`NumberOfDigits`](https://github.com/TheAlgorithms/Java/blob/57878cac555eebf33a86b3d808e7b91f26174458/src/main/java/com/thealgorithms/maths/NumberOfDigits.java) to determine the number of iterations in the loop.

Reply:
Thanks, I don't know about these methods. It's kind of redundant. Because code will look like this:

```
for (int i = 0, exp = 1; i < NumberOfDigits.numberOfDigits(max); i++, exp *= BASE) {
    countingSortByDigit(array, exp);
}
```

But it will improve the readability of the code and its understanding.